module-services/service-cellular/src/SimCard.cpp

```diff
@@ -225,11 +225,17 @@ namespace cellular
                 if (auto state = simState(); state) {
                     handleSimState(*state);
                 }
-            }
-            else if (result != sim::Result::OK) {
-                if (onUnhandledCME)
+            } else if (result != sim::Result::OK) {
+		if (result == sim::Result::SIM_PUKRequired) {
+			// MOS-242: silence this case to avoid a misleading "wrong PUK" pop-up
```

Comment:
This is a hack/workaround for today's cellular message handling: the true cause of this issue lies in doubled PUK request /"need PUK" error messages passed down our system. By making this change we silence one of them, allowing the second one to do its job so that the proper "enter PUK" pop-up is displayed instead of the distorted windows.